runtime/src/qio/qio_formatted.c

```diff
@@ -420,6 +509,8 @@ qioerr qio_channel_read_string(const int threadsafe, const int byteorder, const
     case QIO_BINARY_STRING_STYLE_TOEOF:
       // read until the end of the file.
       // Figure out how many bytes are available.
+      // This case has already been handled above
+      break;
```

Comment:
the lines after this break statement  and before `default:` can not be run now so you should delete them